Fusion in Decoder (FiD) (#310)
primeqa/primeqa: primeqa/pipelines/README.md

largest inland city"""]
answers = reader.apply(question,context)  
print(json.dumps(answers, indent=4))  
```

### Generative FiD Reader

> A Generative Reader takes a question and uses a set of supporting passages to qenerate an answer. Compared to the Extractive Reader, where the answers are usually short spans extracted from the input passages, the Generative Reader generates complex, multi-sentence answers.

Reviewer: ```suggestion
A Generative Reader takes a question and uses a set of supporting passages to generate an answer. In contrast to the Extractive Reader, where the answers are usually short spans extracted from the input passages, the Generative Reader generates complex, multi-sentence answers.
```

            the exhaust of a jet engine condenses on particles, which come from either the 
            surrounding air or the exhaust itself, and freezes, leaving behind a visible trail. 
            The exhaust can also trigger the formation of cirrus by providing ice nuclei 
            when there is an insufficient naturally-occurring supply in the atmosphere. 
            One of the environmental impacts of aviation is that persistent contrails can 
            form into large mats of cirrus, and increased air traffic has been implicated 
            as one possible cause of the increasing frequency and amount of cirrus"""]]
answers = fid_reader.apply(question,context)  
print(json.dumps(answers, indent=4)) 
```

## QA Pipeline

The QA pipeline is used for Open Retrieval Question Answering. Open retrieval systems query large document stores for relevant passages. Long Form Question Answering (LFQA) is a generative task where the retrieved passages are used to generate a complex multi-sentence answer.

In this example we show a QA Pipeline using a ColBERT retriever  and a Fusion in Decoder (FID) generator.

> Instructions to create a ColBERT index and an FiD model for KILT-ELI5 can be found [here]()

Reviewer: missing link

- Step 1:  Initialize the retriever.

```python
retriever = ColBERTRetriever(index_root = index_root, index_name = index_name, collection = collection, max_num_documents = 3)
```

> - Step 2:  Initialize the reader model. You can choose any the generative QA model we currently have ...

Reviewer: ```suggestion
- Step 2:  Initialize the reader model. You can choose any generative QA model we currently have ...
```

largest inland city"""]
answers = reader.apply(question,context)  
print(json.dumps(answers, indent=4))  
```

### Generative FiD Reader

> A Generative Reader takes a question and uses a set of supporting passages to generate an answer. In contrast to the Extractive Reader, where the answers are usually short spans extracted from the input passages, the Generative Reader generates complex, multi-sentence answers.

Reviewer: ```suggestion
A Generative Reader takes a question and a set of supporting passages (FiD usually uses 3 supporting passage) to generate an answer. In contrast to the Extractive Reader, where the answers are usually short spans extracted from the input passages, the Generative Reader generates complex, multi-sentence answers.
```
